### ISSUE TITLE
Autolinks to functions point to the struct with the same name

### ISSUE THREAD

**Issue by yous** [doxygen/doxygen#10349]
Creating a new issue from https://github.com/doxygen/doxygen/issues/9587#issuecomment-1752116942.

**Describe the bug**
When there is a struct and a function with the same name, Doxygen autolinks to the struct, not to the function.

**Expected behavior**
`foo()`, `foo(int, int)`, and `foo(int a, int b)` should be linked to the function. But I'm not sure what should I use to link to the struct.

**Screenshots**
<img width="458" alt="Screenshot 2023-10-09 02 48 12" src="https://github.com/doxygen/doxygen/assets/853977/ea4d9f8d-ffdf-4a61-b50c-e49ec5202f63">

**To Reproduce**
- [SameNameStructAndFunction.tar.gz](https://github.com/doxygen/doxygen/files/12841327/SameNameStructAndFunction.tar.gz)

``` c
/**
 * @file    header.h
 * @brief   description header file
 */

#ifndef HEADER_H
#define HEADER_H

/**
 * @struct foo
 * @brief struct named foo
 *
 * @sa foo()
 */
struct foo {
    int x;
    int y;
};

/**
 * @brief function named foo
 *
 * @sa @link foo @endlink
 */
int foo(int a, int b);

#endif /* HEADER_H */
```

Run `doxygen .`.

**Version**
Both on Intel macOS:
- 1.9.8
- 1.10.0 (73874def7998419647a6281ec4b39c4261d4d18f*)

**Comment by yous** [comment]
Doxygen 1.9.4 effectively distinguish `@link foo @endlink` and `foo()`, and links them to the struct and the function, respectively. When I did `git bisect`, it points to b290399fab92df45376103d7c094f053f24495fa.

**Comment by doxygen** [comment]
@yous Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release is made.

**Comment by yous** [comment]
Hi, it seems the issue is almost resolved, but in Function Documentation, the name of the function has a link to the struct.

<img width="246" alt="Screenshot 2023-11-05 17 14 25" src="https://github.com/doxygen/doxygen/assets/853977/2182149d-e799-418c-b49b-ab981257a90a">

The blue `foo` points to `structfoo.html`. But when there is no struct with the same name, that `foo` won't be linked, so I'm neutral to this.

**Comment by doxygen** [comment]
@yous Good point. Should be fixed with the latest commit.

**Comment by yous** [comment]
Yes, I can confirm it's fixed.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
